src/core_plugins/vega/index.js

```diff
@@ -0,0 +1,24 @@
+export default kibana => new kibana.Plugin({
+  id: 'vega',
+  require: ['elasticsearch'],
+
+  uiExports: {
+    visTypes: [
+      'plugins/vega/vega_type'
+    ],
+    // fixme: For some reason, injectVars() is never called
+    // fixme: once fixed, remove it from src/core_plugins/kibana/inject_vars.js
+    // app: {
+    //   injectVars: (server) => ({ vegaConfig: server.config().get('vega') })
```

Comment:
`app.injectVars` is only called when rendering that specific app, but this plugin doesn't actually define an app. You need `injectDefaultVars`, which will be called and can return vars that will be injected in every app.